libsolidity/ast/Types.cpp

```diff
@@ -40,10 +40,35 @@
 
 #include <limits>
 
+#include <boost/multiprecision/cpp_dec_float.hpp>
+
 using namespace std;
 using namespace dev;
 using namespace dev::solidity;
 
+namespace
+{
+
+bool fitsPrecision(bigint const _base, bigint const _exp)
+{
+	using boost::multiprecision::log10;
+	using boost::multiprecision::ceil;
+	using bigfloat = boost::multiprecision::number<boost::multiprecision::cpp_dec_float<50>>;
+
+	size_t const bitsMax = 4096;
+	bigint const baseMax = (bigint(2) << bitsMax) - 1;
+	if(_base > baseMax)
+		return false;
+
+	bigfloat bitsNeeded = bigfloat(_exp) * ceil(log10(bigfloat(_base + 1)));
```

Comment:
Maybe ``log2`` instead of ``log10``?

Reply:
That was a defect ;) ``log10`` is what I wanted but this formula calculated the number of digits and not the number of bits. I've added another formula which estimates the minimum number of bits needed.